builtin/sampled_dataset.cc

```diff
@@ -299,29 +299,34 @@ SampledDataset(MldbServer * owner,
 {
      SampledDatasetConfig config;
 
-    if(!options.isRow())
-        throw ML::Exception("options should be a row");
+    if(options.isRow()) {
+        for(auto elem : options.getRow()) {
+            const ColumnName& columnName = std::get<0>(elem);
 
-    for(auto elem : options.getRow()) {
-        const ColumnName& columnName = std::get<0>(elem);
-
-        if (columnName == ColumnName("rows")) {
-            config.rows = std::get<1>(elem).toInt();
-        }
-        else if (columnName == ColumnName("fraction")) {
-            config.fraction = std::get<1>(elem).toDouble();
-        }
-        else if (columnName == ColumnName("withReplacement")) {
-            config.withReplacement = std::get<1>(elem).asBool();
-        }
-        else if (columnName == ColumnName("seed")) {
-            config.seed = std::get<1>(elem).toInt();
-        }
-        else {
-            auto expVal2 = std::get<1>(elem);
-            throw ML::Exception("unknown option: '"+columnName.toString()+"'");
+            if (columnName == ColumnName("rows")) {
+                config.rows = std::get<1>(elem).toInt();
+            }
+            else if (columnName == ColumnName("fraction")) {
+                config.fraction = std::get<1>(elem).toDouble();
+            }
+            else if (columnName == ColumnName("withReplacement")) {
+                config.withReplacement = std::get<1>(elem).asBool();
+            }
+            else if (columnName == ColumnName("seed")) {
+                config.seed = std::get<1>(elem).toInt();
+            }
+            else {
+                auto expVal2 = std::get<1>(elem);
+                throw ML::Exception("unknown option: '"+columnName.toString()+"'");
```

Comment:
I know this is not new code but I really feel we should not reference an internal name, here `option`, when returning an error.  `params` is probably more appropriate since it is mentioned in the `sampled` dataset documentation.  Also, we should mention that this is returned by the `sample` function.